### ISSUE TITLE
Throw error when creating histogram with too many axes

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/boost-histogram#128]
The following test succeeds, but it should not:

```python
def test_out_of_limit_axis():
    for i in range(36):
        ax = (bh.axis.regular(1, -1, 1, underflow=False, overflow=False) for a in range(i))
        bh.histogram(*ax)
```

**Comment by HDembinski** [comment]
I will fix this in boost::histogram, but it should also be caught in the histogram init.

**Comment by henryiii** [comment]
Fixed in #126. It would be nice to have it fixed in Boost.Histogram, so let me know if you would like me to open an issue there as a reminder. This should almost never happen, so it's not a high priority.

**Comment by HDembinski** [comment]
I opened an issue for it in boost::histogram, thanks for spotting this.
